cmd/modern/root/install/mssql-base.go

```diff
@@ -408,7 +410,14 @@ func (c *MssqlBase) validateUsingUrlExists() {
 
 	// At the moment we only support attaching .bak files, but we should
 	// support .bacpacs and .mdfs in the future
-	if _, file := filepath.Split(c.usingDatabaseUrl); filepath.Ext(file) != ".bak" {
+	databaseName := parseDbName(c.usingDatabaseUrl)
+	databaseUrl := c.usingDatabaseUrl
+	if databaseName != "" {
+		argLen := len(c.usingDatabaseUrl)
+		dbNameLen := len(databaseName)
+		databaseUrl = c.usingDatabaseUrl[0:(argLen - dbNameLen - 1)]
+	}
+	if _, file := filepath.Split(databaseUrl); filepath.Ext(file) != ".bak" {
```

Comment:
>filepath.Split(databaseUrl);  [](http://example.com/codeflow?start=15&length=29)

shouldn't we be using the last segment of url.Path as the value of file? The file name in the URL is going to be url encoded.